HW2/P2/parallel_vector.pyx

```diff
@@ -81,10 +81,14 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
                if counts[src[idx]] > 0:
+                   acquire(&locks[dest[idx]])
```

Comment:
Interesting way to avoid deadlocking and double locking, by having no thread acquire more than 1 lock at a time.